config/initializers/content_security_policy.rb

```diff
@@ -2,17 +2,34 @@
 # For further information see the following documentation
 # https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Content-Security-Policy
 
-# Rails.application.config.content_security_policy do |p|
-#   p.default_src :self, :https
-#   p.font_src    :self, :https, :data
-#   p.img_src     :self, :https, :data
-#   p.object_src  :none
-#   p.script_src  :self, :https
-#   p.style_src   :self, :https, :unsafe_inline
-#
-#   # Specify URI for violation reports
-#   # p.report_uri "/csp-violation-report-endpoint"
-# end
+if Rails.env.production?
+  assets_host = Rails.configuration.action_controller.asset_host || "https://#{ENV['WEB_DOMAIN'] || ENV['LOCAL_DOMAIN']}"
+  data_hosts = [assets_host]
+
+  if ENV['S3_ENABLED'] == 'true'
+    attachments_host = ENV['S3_ALIAS_HOST'] || ENV['S3_CLOUDFRONT_HOST'] || ENV['S3_HOSTNAME'] || "s3-#{ENV['S3_REGION'] || 'us-east-1'}.amazonaws.com"
```

Comment:
What is the trailing `'us-east-1'}.amazonaws.com"` for?

Reply:
I have already merged a fix for this that does not rely on composing these URLs manually, by whitelisting https. Whitelisting https for images and videos is better than no CSP at all. At the same time, predicting which URLs it's gonna be is very complicated, because there's S3, Wasabi, Google Cloud, Swift, etc, and afaik no way to get that out of Paperclip without reading the environment variables again and essentially duplicating its logic. So, imo not worth it. Just whitelist https.